logging/troubleshooting/cluster-logging-troubleshooting-for-critical-alerts.adoc

```diff
@@ -0,0 +1,538 @@
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/settings?pretty -X PUT -d '{"persistent": {"cluster.routing.allocation.enable":"all"}}'
+----
+
+. Check which indices are still red.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+. If any indices are still red, try to clear them by performing the following steps.
+
+.. Clear the cache.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_cache/clear?pretty
+----
+
+.. Increase the max allocation retries.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.allocation.max_retries":10}'
+----
+
+.. Delete all the scroll items.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_search/scroll/_all -X DELETE
+----
+
+.. Increase the timeout.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.unassigned.node_left.delayed_timeout":"10m"}'
+----
+
+. If the preceding steps do not clear the red indices, delete the indices individually.
+
+.. Identify the red index name.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+.. Delete the red index.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_red_index_name> -X DELETE
+----
+
+. If there are no red indices and the cluster status is red, check for a continuous heavy processing load on a data node.
+
+.. Check if the Elasticsearch JVM Heap usage is high.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c <elasticsearch_pod_name> -- es_util --query=_nodes/stats?pretty
+----
++
+In the command output, review the `node_name.jvm.mem.heap_used_percent` field to determine the JVM Heap usage.
+
+.. Check for high CPU utilization.
+
+.Additional resources
+
+* See "Free up or increase disk space" in the Elasticsearch topic, link:https://www.elastic.co/guide/en/elasticsearch/reference/7.13/fix-common-cluster-issues.html#fix-red-yellow-cluster-status[Fix a red or yellow cluster status].
+
+[id="elasticsearch-cluster-health-is-yellow"]
+== Elasticsearch Cluster Health is Yellow
+
+Replica shards for at least one primary shard are not allocated to nodes.
+
+.Troubleshooting
+
+. Check the disk space of the Elasticsearch node.
+
+. Increase the node count or the disk space of existing nodes by adjusting `nodeCount` or `storage.size` in the `ClusterLogging` CR.
```

Comment:
Adjusting `storage.size` in clusterlogging is not supported, this action won't make any change to the ES cluster. Details are in https://issues.redhat.com/browse/LOG-920.

Reply:
Yes, @QiaolingTang is right. This action won't work.
@rolfedh we haven't still worked on `warning` alerts hence you can remove these troubleshooting steps.

Reply:
@sasagarw It seems like, based on @QiaolingTang's statement, I should simply remove the previous step and adjust the following phrase as shown here:
```suggestion
. Increase the node count by adjusting `nodeCount` in the `ClusterLogging` CR.
```
Otherwise, if the scope of the issue is larger than this, let me know which troubleshooting steps should be removed.

Reply:
I guess we can let it be like this based on your updated statement.
When we will plan to document the steps for `Elasticsearch Cluster Health is Yellow`, we will update the complete troubleshooting steps.